src/api-engine/api/routes/chaincode/views.py

```
@@ -52,13 +57,19 @@ def _read_cc_pkg(self, pk, filename, ccpackage_path):
             meta_path = os.path.join(ccpackage_path, "metadata.json")
             # extract metadata file
             with tarfile.open(os.path.join(ccpackage_path, filename)) as tared_file:
-                metadata_file = tared_file.getmember("metadata.json")
-                tared_file.extract(metadata_file, path=ccpackage_path)
-
-            with open(meta_path, 'r') as f:
-                metadata = json.load(f)
-                language = metadata["type"]
-                label = metadata["label"]
+                metadata_file = None
+                for member in tared_file.getmembers():
+                    if member.name.endswith("metadata.json"):
+                        metadata_file = member
+                        break
+
+                if metadata_file is not None:
+                    # Extract the metadata file
+                    metadata_content = tared_file.extractfile(
+                        metadata_file).read().decode("utf-8")
+                    metadata = json.loads(metadata_content)
+                    language = metadata["type"]
+                    label = metadata["label"]
```

If the file is not found, it's better to report the error. That means could you deal with the situation when metadata_file is None?

---

Sure, I'll push these changes soon